### ISSUE TITLE
Discussion: (batch_size * loss).backward()

### ISSUE THREAD

**Issue by MattKleinsmith** [irrmnv/pytorch-ieee-cmi#1]
I notice this code does this:

```python
optimizer.zero_grad()
batch_size = inputs.size(0)
(batch_size * loss).backward()
optimizer.step()
```
[L81](https://github.com/irrmnv/pytorch-ieee-cmi/blob/master/train_utils.py#L81)

instead of this:

```python
optimizer.zero_grad()
loss.backward()
optimizer.step()
```

Can someone say something about this choice?

It seems equivalent to multiplying the gradients or learning rate by the batch size, but I'm not sure. I would appreciate anyone's thoughts.
